docs/proposed/psa-conditional-inclusion-c.md

```diff
@@ -107,9 +115,39 @@ These symbols are not part of the public interface of Mbed TLS towards applicati
 
 ### Architecture of symbol definitions
 
-#### Definition of internal inclusion symbols
+#### New-style definition of configuration symbols
+
+When `MBEDTLS_PSA_CRYPTO_CONFIG` is set, the header file `mbedtls/config.h` needs to define all the `MBEDTLS_xxx_C` configuration symbols, including the ones deduced from the PSA Crypto configuration. It does this by including the new header file **`mbedtls/config_psa.h`**, which defines the `MBEDTLS_PSA_BUILTIN_xxx` symbols and deduces the corresponding `MBEDTLS_xxx_C` (and other) symbols.
+
+`mbedtls/config_psa.h` includes `psa/crypto_config.h`, the user-editable file that defines application requirements.
+
+#### New-style definition of configuration symbols
+
+When `MBEDTLS_PSA_CRYPTO_CONFIG` is not set, the configuration of Mbed TLS works as before, and the inclusion of non-PSA code only depends on `MBEDTLS_xxx` symbols defined (or not) in `mbedtls/config.h`. Furthermore, a new header file **`mbedtls/config_psa.h`** deduces PSA configuration symbols (`PSA_WANT_xxx`, `MBEDTLS_PSA_BUILTIN_xxx`) from classic configuration symbols (`MBEDTLS_xxx`).
```

Comment:
“A new header file” is correct, why change to “the”?

You objected to “in that case” in an earlier version, and since “that case” is now the whole section it would be more out of place now.

Reply:
About "the" it is because in the previous section config_psa.h is introduced as "the new header file **`mbedtls/config_psa.h`**" thus using "a" now is a bit weird to me, a bit like it is another file. I think there is some kind of a rule that when you first introduce something new you use "a" and after you use "the" thus here I'd rather do it the other way around: "a" in the previous paragraph and "the" here. But I can be wrong.

Reply:
“A” is correct here since this sentence introduces the new file. But “the” is also correct, so I've changed it to “the”.